lib/ddtrace/auto_instrument_base.rb

```diff
@@ -0,0 +1,15 @@
+module Datadog
+  # base methods stubbed for adding auto instrument extensions
+  module AutoInstrumentBase
+
+    def self.included(base)
+      base.send(:extend, InstanceMethods)
+      base.send(:include, InstanceMethods)
+    end
```

Comment:
Any particular reason to indirectly extend `InstanceMethods`, instead of defining `AutoInstrumentBase` as:
```ruby
module Datadog
  # base methods stubbed for adding auto instrument extensions
  module AutoInstrumentBase
    # stubbed methods for adding auto instrument
    def add_auto_instrument; end
  end
end
```

and then `extend`ing it on the base module:
```ruby
module Datadog
  extend AutoInstrumentBase
```

Reply:
no not really, i was following a pattern i'd seen with some of the config modules, but this is far simpler so can update